crates/sage/src/scoring.rs

```diff
@@ -116,8 +118,21 @@ pub struct Feature {
 
     pub ms2_intensity: f32,
     pub ms1_intensity: f32,
+
+    pub fragments: Fragments
 }
 
+/// Matching Fragment details
+#[derive(Serialize, Default, Clone, Debug)]
+pub struct Fragments {
```

Comment:
I love the data oriented/columnar design here. Very clever!

Reply:
Thank you for your feedback.